### PR TITLE
Migrate HoldToAuthorizeButton to use Reanimated 2

### PR DESCRIPTION
**Pull request by jkadamczyk**
Fixes RNBW-3390

## What changed (plus any additional context for devs)

* Rewrote files I worked on to TypeScript + fixed some JS files to play better with typescript
* Rewrote class component to function component to be able to use Reanimated 2
* Rewrote Reanimated 1 animations to Reanimated 2

## PoW (screenshots / screen recordings)
Before:
https://streamable.com/6uzmyj
After light:
https://streamable.com/7mpc1x
After dark:
https://streamable.com/zhe4cp

## Dev checklist for QA: what to test
* Try sending a transaction on iOS phone with face id
* Hold the button to authorise and watch the loading progress animation

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels?
- [x] Added e2e tests? if not please specify why – animations, and there wasn't any in the first place
- [x] If you added new files, did you update the CODEOWNERS file?

### CODE REVIEW

**Comment by jkadamczyk** [comment]
ENS Profiles folks, if you can merge this branch on top of your ENS profiles branch, please check if the hold to authorise is still working as desired, the changes were under the hood so it should be okay but it's better to check.

CC @jxom @estebanmino @pugson @benisgold

File: src/components/buttons/hold-to-authorize/HoldToAuthorizeButtonContent.tsx
Comment (brunobar79) [inline src/components/buttons/hold-to-authorize/HoldToAuthorizeButtonContent.tsx:81]: same

**Comment by jkadamczyk** [comment]
@brunobar79 yeah I will fix those todo comments, I forgot about them. I was thinking about typing colors anyway. It will take some time but will definitely be useful.

**Review by tchayen — APPROVED** [review]
🚀

**Comment by brunobar79** [comment]
linter complaining about imports, otherwise LGTM!

**Comment by jkadamczyk** [comment]
e2e failing will investigate tomorrow

**Comment by jkadamczyk** [comment]
It's failing because of Sandbox, ,it will start building once we merge sandbox changes

File: CODEOWNERS
Comment (osdnk) [inline CODEOWNERS:31]: I have some opinions about this https://rainbowhaus.slack.com/archives/CGUDDJYNL/p1652689873841879

**Review by osdnk — CHANGES_REQUESTED** [review]
I observe differences, not sure if expected, rather no:
After your changes: 
https://streamable.com/ztlest
Before:
https://streamable.com/ird35d

Observe the spinner after the interaction.

**Comment by jkadamczyk** [comment]
> I observe differences, not sure if expected, rather no: After your changes: [streamable.com/ztlest](https://streamable.com/ztlest) Before: [streamable.com/ird35d](https://streamable.com/ird35d)
> 
> Observe the spinner after the interaction.

Yeah the difference was intentional
https://rainbowhaus.slack.com/archives/CGUNEHL9K/p1651839295699659

File: CODEOWNERS
Comment (jkadamczyk) [reply CODEOWNERS:31]: You're right about it I will leave both you and Terry there next to me.

**Review by estebanmino — APPROVED** [review]
working on profiles! 🚀

File: src/components/buttons/hold-to-authorize/constants.ts
Comment (osdnk) [reply src/components/buttons/hold-to-authorize/constants.ts:8]: please removethe comment

File: src/components/icons/svg/ProgressIcon.tsx
Comment (osdnk) [inline src/components/icons/svg/ProgressIcon.tsx:15]: Do you use values 0 to 1 that you convert to degrees and then convert degrees to radians? Can't you just use radians directly?

**Review by osdnk — APPROVED** [review]
hot 🥵  🔥

File: src/components/buttons/hold-to-authorize/types/HoldToAuthorizeBaseProps.ts
Comment (markdalgleish) [inline src/components/buttons/hold-to-authorize/types/HoldToAuthorizeBaseProps.ts:16]: Can you make this type more specific?

Comment (markdalgleish) [inline src/components/buttons/hold-to-authorize/types/HoldToAuthorizeBaseProps.ts:16]: Can you make this type more specific?

File: src/components/buttons/hold-to-authorize/types/HoldToAuthorizeBaseProps.ts
Comment (markdalgleish) [inline src/components/buttons/hold-to-authorize/types/HoldToAuthorizeBaseProps.ts:19]: Same as above - can this type be more specific?

Comment (markdalgleish) [inline src/components/buttons/hold-to-authorize/types/HoldToAuthorizeBaseProps.ts:19]: Same as above - can this type be more specific?

**Review by markdalgleish — APPROVED** [review]
Just a couple of notes about potentially making the types more specific, but I'm happy to leave it to you from here.

**Pull request merged**
